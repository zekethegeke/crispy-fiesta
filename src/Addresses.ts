import { WekePageRef } from 'weke-plex';

export interface Address {
    id: number,
    first: string,
    last: string,
    email: string,
    gender: string,
    ip: string
}

export class Addresses {
    readonly males: Address[];
    readonly females: Address[];

    constructor(readonly all: Address[]) {
        this.males = all.filter(it => it.gender === "Male");
        this.females = all.filter(it => it.gender === "Female");
    }

    static ofJson(json: string) {
        return new Addresses(JSON.parse(json) as Address[]);
    }
}

export const addressesPage = new WekePageRef("AddressesPage");
