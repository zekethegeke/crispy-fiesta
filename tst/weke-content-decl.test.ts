import { wrap } from 'jest-snapshot-serializer-raw';
import { Address, Addresses, addressesPage } from '../src/Addresses';
import { block, group, h3, span, uml, list1, list2, tableByMap, tableOf, tableByColumns } from 'weke-plex';

const addresses = (new Addresses(require("./test-addresses.json")));
const males = addresses.males.map(it => `"${it.first}" -> "Male"`);

test('yet another mark around language', () => {
    const content = [
        h3 `Purpose`,
        block `
            Our purpose is to test the content.`,
        h3 `//${addressesPage}//`,
        uml `
            (*) - (*)
            ${males}`,
    ].join("\n");
    expect(wrap(content)).toMatchSnapshot();
});

test('yet another mark around list', () => {
    const dotComs = addresses.all.filter((it) => it.email.endsWith('.com'));
    const ofMales = (addy: Address) => males.filter((it) => it.includes(addy.first));

    const content = [
        h3 `Lists`,
        list1(males, (male) => 
            span `* each male: ${male}`),

        list2(dotComs, (dotCom) => 
            [ span `* each email: ${dotCom.email}`, 
                ofMales(dotCom)],
            (male) => 
                span `** each male: ${male}`),
    ].join("\n");
    expect(wrap(content)).toMatchSnapshot();
});

test('yet another mark around list', () => {
    const dotComs = addresses.all.filter((it) => it.email.endsWith('.com'))
        .slice(0, 10);

    const content = [
        h3 `tableOf`,
        tableOf(dotComs),

        h3 `tableOf filtered`,
        tableOf(dotComs, ['first', 'email']),

        h3 `tableByMap`,
        //  tableOf
        tableByMap(dotComs, (it) => {
            return { 
                "//First Name//": 
                    span `cell: ${it.first}`,
                "Last Name":
                    span `cell: ${it.last}`,
            }
        }),

        h3 `tableByColumns`,
        tableByColumns({
            rows: dotComs, 
            columns: [{   
                key: 'first', 
                headOf: (key) => `//${key.toUpperCase()}//`,
                cellOf: (data) => span `cell: ${data.value}`
            },{   
                key: 'last', 
                headOf: (key, colIndex) => `//${key.toUpperCase()}#${colIndex}//`,
                cellOf: (data) =>  group `
                    multiline: ${data.value}
                    ${data.rowIndex}:${data.colIndex}`
            },{   
                key: 'last', 
                headOf: (key, colIndex) => `//${key.toUpperCase()}#${colIndex}//`,
                cellOf: (data) =>  group `
                    multiline: ${data.value}
                    ${data.rowIndex}:${data.colIndex}`
            },]
        }),
        ].join("\n");
    expect(wrap(content)).toMatchSnapshot();
});
