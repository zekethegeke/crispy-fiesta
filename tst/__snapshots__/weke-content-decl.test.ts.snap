// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yet another mark around language 1`] = `
=== Purpose ===
Our purpose is to test the content.
=== //[[AddressesPage]]// ===
{{plantUML}}
@startuml
(*) - (*)
"Tannie" -> "Male"
"Lauren" -> "Male"
"Ervin" -> "Male"
"Hersch" -> "Male"
"Darnall" -> "Male"
"Moise" -> "Male"
"Dunstan" -> "Male"
"Klemens" -> "Male"
"Lancelot" -> "Male"
"Arlan" -> "Male"
"Clemmy" -> "Male"
"Tyrus" -> "Male"
"Colas" -> "Male"
"Dun" -> "Male"
"Demetri" -> "Male"
"Arron" -> "Male"
"Purcell" -> "Male"
"Nigel" -> "Male"
"Pryce" -> "Male"
"Gav" -> "Male"
"Piotr" -> "Male"
"Rickey" -> "Male"
"Wolf" -> "Male"
"Derron" -> "Male"
"Artemas" -> "Male"
"Ernestus" -> "Male"
"Bordy" -> "Male"
"Anatollo" -> "Male"
"Thornie" -> "Male"
"Thorin" -> "Male"
"Neddy" -> "Male"
"Germaine" -> "Male"
"Pablo" -> "Male"
"Aldrich" -> "Male"
"Dale" -> "Male"
"Nappie" -> "Male"
"Nico" -> "Male"
"Shaun" -> "Male"
"Josias" -> "Male"
"Rodolfo" -> "Male"
"Duke" -> "Male"
"Keary" -> "Male"
"Bond" -> "Male"
"Colan" -> "Male"
"Alwin" -> "Male"
"Jackie" -> "Male"
"Christian" -> "Male"
@enduml
{{/plantUML}}
`;

exports[`yet another mark around list 1`] = `
=== Lists ===
* each male: "Tannie" -> "Male"
* each male: "Lauren" -> "Male"
* each male: "Ervin" -> "Male"
* each male: "Hersch" -> "Male"
* each male: "Darnall" -> "Male"
* each male: "Moise" -> "Male"
* each male: "Dunstan" -> "Male"
* each male: "Klemens" -> "Male"
* each male: "Lancelot" -> "Male"
* each male: "Arlan" -> "Male"
* each male: "Clemmy" -> "Male"
* each male: "Tyrus" -> "Male"
* each male: "Colas" -> "Male"
* each male: "Dun" -> "Male"
* each male: "Demetri" -> "Male"
* each male: "Arron" -> "Male"
* each male: "Purcell" -> "Male"
* each male: "Nigel" -> "Male"
* each male: "Pryce" -> "Male"
* each male: "Gav" -> "Male"
* each male: "Piotr" -> "Male"
* each male: "Rickey" -> "Male"
* each male: "Wolf" -> "Male"
* each male: "Derron" -> "Male"
* each male: "Artemas" -> "Male"
* each male: "Ernestus" -> "Male"
* each male: "Bordy" -> "Male"
* each male: "Anatollo" -> "Male"
* each male: "Thornie" -> "Male"
* each male: "Thorin" -> "Male"
* each male: "Neddy" -> "Male"
* each male: "Germaine" -> "Male"
* each male: "Pablo" -> "Male"
* each male: "Aldrich" -> "Male"
* each male: "Dale" -> "Male"
* each male: "Nappie" -> "Male"
* each male: "Nico" -> "Male"
* each male: "Shaun" -> "Male"
* each male: "Josias" -> "Male"
* each male: "Rodolfo" -> "Male"
* each male: "Duke" -> "Male"
* each male: "Keary" -> "Male"
* each male: "Bond" -> "Male"
* each male: "Colan" -> "Male"
* each male: "Alwin" -> "Male"
* each male: "Jackie" -> "Male"
* each male: "Christian" -> "Male"
* each email: dpinchon1@salon.com
* each email: escedall2@hc360.com
* each email: varlidge5@fotki.com
* each email: cletessier7@meetup.com
* each email: hscrivinor8@yellowbook.com
** each male: "Hersch" -> "Male"
* each email: ematfieldb@webs.com
* each email: ecasarinie@meetup.com
* each email: dbarszczewskif@buzzfeed.com
** each male: "Dunstan" -> "Male"
* each email: bmclugaishg@mashable.com
* each email: kgalieroh@lycos.com
** each male: "Klemens" -> "Male"
* each email: mchomickii@nba.com
* each email: ldillwayj@ft.com
** each male: "Lancelot" -> "Male"
* each email: hhalmkink@linkedin.com
* each email: abarfootl@tinyurl.com
** each male: "Arlan" -> "Male"
* each email: cdoyleym@xinhuanet.com
** each male: "Clemmy" -> "Male"
* each email: todonoghuen@blogtalkradio.com
** each male: "Tyrus" -> "Male"
* each email: cmccullumo@paypal.com
** each male: "Colas" -> "Male"
* each email: dcastaneq@alexa.com
** each male: "Demetri" -> "Male"
* each email: dmacbarronu@cbslocal.com
* each email: ntrevanw@java.com
** each male: "Nigel" -> "Male"
* each email: pgyenesy@deviantart.com
** each male: "Pryce" -> "Male"
* each email: arichardt10@omniture.com
* each email: tfeare12@1688.com
* each email: dmiddler13@blogs.com
* each email: rlogie17@macromedia.com
* each email: srowan18@meetup.com
* each email: rkenderdine19@pcworld.com
** each male: "Rickey" -> "Male"
* each email: eormrod1a@accuweather.com
* each email: ahovel1f@bing.com
* each email: dkesterton1g@hp.com
** each male: "Derron" -> "Male"
* each email: cshearme1h@netlog.com
* each email: cbaile1i@youku.com
* each email: kcrepin1j@omniture.com
* each email: astellman1k@livejournal.com
** each male: "Artemas" -> "Male"
* each email: jewert1m@examiner.com
* each email: bletts1n@devhub.com
** each male: "Bordy" -> "Male"
* each email: tcluckie1r@webmd.com
** each male: "Thorin" -> "Male"
* each email: acholwell1s@nytimes.com
* each email: pdruce1w@studiopress.com
** each male: "Pablo" -> "Male"
* each email: akilgour1x@meetup.com
** each male: "Aldrich" -> "Male"
* each email: ezarfai1z@tinyurl.com
* each email: nrentenbeck20@ibm.com
** each male: "Nappie" -> "Male"
* each email: tantoniat21@tinypic.com
* each email: ncass23@canalblog.com
** each male: "Nico" -> "Male"
* each email: kscamerden25@newsvine.com
* each email: gleonard26@go.com
* each email: pschachter28@blogtalkradio.com
* each email: jtatlowe29@ft.com
** each male: "Josias" -> "Male"
* each email: hosheeryne2a@sogou.com
* each email: rfargie2b@ted.com
** each male: "Rodolfo" -> "Male"
* each email: acrumpton2c@aol.com
* each email: kbacks2e@yelp.com
** each male: "Keary" -> "Male"
* each email: bhuckstepp2f@bandcamp.com
** each male: "Bond" -> "Male"
* each email: cmaccaffery2g@163.com
** each male: "Colan" -> "Male"
* each email: lcochran2h@salon.com
* each email: mmacknockiter2j@feedburner.com
* each email: aklug2k@xing.com
** each male: "Alwin" -> "Male"
* each email: dcaherny2n@ifeng.com
* each email: rleupoldt2p@opera.com
* each email: temblem2r@chicagotribune.com
`;

exports[`yet another mark around list 2`] = `
=== tableOf ===
|= id|= first|= last|= email|= gender|= ip
| 2| Drucill| Pinchon| dpinchon1@salon.com| Female| 98.255.222.40
| 3| Ebony| Scedall| escedall2@hc360.com| Female| 41.149.143.137
| 6| Val| Arlidge| varlidge5@fotki.com| Female| 71.204.239.240
| 8| Cloris| Letessier| cletessier7@meetup.com| Female| 15.251.243.105
| 9| Hersch| Scrivinor| hscrivinor8@yellowbook.com| Male| 159.237.108.139
| 12| Elisabet| Matfield| ematfieldb@webs.com| Female| 172.59.36.249
| 15| Elaina| Casarini| ecasarinie@meetup.com| Female| 247.129.72.180
| 16| Dunstan| Barszczewski| dbarszczewskif@buzzfeed.com| Male| 116.235.250.253
| 17| Beitris| McLugaish| bmclugaishg@mashable.com| Female| 115.83.160.125
| 18| Klemens| Galiero| kgalieroh@lycos.com| Male| 26.190.54.239

=== tableOf filtered ===
|= first|= email
| Drucill| dpinchon1@salon.com
| Ebony| escedall2@hc360.com
| Val| varlidge5@fotki.com
| Cloris| cletessier7@meetup.com
| Hersch| hscrivinor8@yellowbook.com
| Elisabet| ematfieldb@webs.com
| Elaina| ecasarinie@meetup.com
| Dunstan| dbarszczewskif@buzzfeed.com
| Beitris| bmclugaishg@mashable.com
| Klemens| kgalieroh@lycos.com

=== tableByMap ===
|= //First Name//|= Last Name
| cell: Drucill| cell: Pinchon
| cell: Ebony| cell: Scedall
| cell: Val| cell: Arlidge
| cell: Cloris| cell: Letessier
| cell: Hersch| cell: Scrivinor
| cell: Elisabet| cell: Matfield
| cell: Elaina| cell: Casarini
| cell: Dunstan| cell: Barszczewski
| cell: Beitris| cell: McLugaish
| cell: Klemens| cell: Galiero

=== tableByColumns ===
|= //FIRST//|= //LAST#1//|= //LAST#2//
| cell: Drucill| (((multiline: Pinchon
0:1
)))| (((multiline: Pinchon
0:2
)))
| cell: Ebony| (((multiline: Scedall
1:1
)))| (((multiline: Scedall
1:2
)))
| cell: Val| (((multiline: Arlidge
2:1
)))| (((multiline: Arlidge
2:2
)))
| cell: Cloris| (((multiline: Letessier
3:1
)))| (((multiline: Letessier
3:2
)))
| cell: Hersch| (((multiline: Scrivinor
4:1
)))| (((multiline: Scrivinor
4:2
)))
| cell: Elisabet| (((multiline: Matfield
5:1
)))| (((multiline: Matfield
5:2
)))
| cell: Elaina| (((multiline: Casarini
6:1
)))| (((multiline: Casarini
6:2
)))
| cell: Dunstan| (((multiline: Barszczewski
7:1
)))| (((multiline: Barszczewski
7:2
)))
| cell: Beitris| (((multiline: McLugaish
8:1
)))| (((multiline: McLugaish
8:2
)))
| cell: Klemens| (((multiline: Galiero
9:1
)))| (((multiline: Galiero
9:2
)))

`;
